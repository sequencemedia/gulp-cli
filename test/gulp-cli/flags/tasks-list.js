'use strict'

const { exec } = require('node:child_process')
const { readFile } = require('node:fs/promises')

const chai = require('chai')
const {
  expect
} = chai

const TASKS_PATTERN = /Label of node [^,']*\n/

const TIMESTAMP_PATTERN = /\[.*\]/

describe('--tasks-list', () => {
  it('prints the tasks', (done) => {
    exec('node ./bin/gulp --tasks-list --cwd ./test/mocks/flags/tasks-list', async (e, stdout) => {
      try {
        expect(e)
          .to.be.null

        expect(stdout)
          .to.match(TASKS_PATTERN)

        const [match] = stdout.match(TASKS_PATTERN)

        const TASKS_FIXTURE = await readFile('./test/expected/flags/tasks-list.txt')

        expect(match.replace(new RegExp(TIMESTAMP_PATTERN, 'g'), ''))
          .to.equal(TASKS_FIXTURE.toString())

        done()
      } catch (e) {
        done(e)
      }
    })
  })

  describe('--compact-tasks', () => {
    it('prints the tasks', (done) => {
      exec('node ./bin/gulp --tasks-list --compact-tasks --cwd ./test/mocks/flags/tasks-list', async (e, stdout) => {
        try {
          expect(e)
            .to.be.null

          expect(stdout)
            .to.match(TASKS_PATTERN)

          const [match] = stdout.match(TASKS_PATTERN)

          const TASKS_FIXTURE = await readFile('./test/expected/flags/tasks-list/compact-tasks.txt')

          expect(match.replace(new RegExp(TIMESTAMP_PATTERN, 'g'), ''))
            .to.equal(TASKS_FIXTURE.toString())

          done()
        } catch (e) {
          done(e)
        }
      })
    })
  })
})
