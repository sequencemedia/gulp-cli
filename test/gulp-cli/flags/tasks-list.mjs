import {
  exec
} from 'node:child_process'

import {
  readFile
} from 'node:fs/promises'

import {
  expect
} from 'chai'

const TASKS_PATTERN = /Label of node [^,']*\n/

const TIMESTAMP_PATTERN = /\[.*\]/

describe('--tasks-list', () => {
  it('prints the tasks', (done) => {
    exec('node ./bin/gulp.mjs --tasks-list --cwd ./test/mocks/flags/tasks-list', async (e, stdout) => {
      try {
        expect(e)
          .to.be.null

        expect(stdout)
          .to.match(TASKS_PATTERN)

        const [match] = stdout.match(TASKS_PATTERN)

        const TASKS_FIXTURE = await readFile('./test/expected/flags/tasks-list.txt')

        expect(match.replace(new RegExp(TIMESTAMP_PATTERN, 'g'), ''))
          .to.equal(TASKS_FIXTURE.toString())

        done()
      } catch (e) {
        done(e)
      }
    })
  })

  describe('--compact-tasks', () => {
    it('prints the tasks', (done) => {
      exec('node ./bin/gulp.mjs --tasks-list --compact-tasks --cwd ./test/mocks/flags/tasks-list', async (e, stdout) => {
        try {
          expect(e)
            .to.be.null

          expect(stdout)
            .to.match(TASKS_PATTERN)

          const [match] = stdout.match(TASKS_PATTERN)

          const TASKS_FIXTURE = await readFile('./test/expected/flags/tasks-list/compact-tasks.txt')

          expect(match.replace(new RegExp(TIMESTAMP_PATTERN, 'g'), ''))
            .to.equal(TASKS_FIXTURE.toString())

          done()
        } catch (e) {
          done(e)
        }
      })
    })
  })
})
