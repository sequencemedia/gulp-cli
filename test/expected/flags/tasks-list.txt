Label of node 3
Label of node 2
Label of node 1
